### PR TITLE
Hardcoded strings replaced with resource variables

### LINKED ISSUE #13: Replace hardcoded string with resource variables
**I'm submitting a ...** 

- [ ] bug report
- [x] feature request

**Current behavior:**

All the textual content and the colors are hardcoded everywhere.

**Expected behavior:**

This should be placed inside `res` folder and should be referenced accordingly to support localization.

**Other information:**

### PR DESCRIPTION
**Pull request by adit19shah**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Build is passing successfully
- [x] All tests are passing 
- [ ] PR is a breaking change (might cause older versions to break)

## Fixes issue

 

Fixes: #13 

## Pull request type

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- All the textual content and the colors were hardcoded everywhere.

## What is the new behavior?

- The strings are placed inside the res folder and referenced accordingly to support localization.

## Other information

None

### CODE REVIEW

File: searchwidget/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -52,4 +52,6 @@
     <color name="blue_light">#01adff</color>
     <color name="blue_dark">#3369e6</color>
 
+    <color name="last_result_color">@android:color/black</color>
```
Comment (harsh-2711) [inline searchwidget/src/main/res/values/colors.xml]: keep the color name as it is in the `name` field value. See the above lines for reference.

File: searchwidget/src/main/java/com/harsh/searchwidget/SearchBar.java [hunk 2]
```diff
@@ -252,7 +252,7 @@ private void init(AttributeSet attributeSet) {
         navIcon.setOnClickListener(this);
 
         // Placeholder
-        placeholderText = "Search Widget";
+        placeholderText = "@string/app_name";
```
Comment (harsh-2711) [inline searchwidget/src/main/java/com/harsh/searchwidget/SearchBar.java]: Don't give placeholder text the same value as app name. The app name might change in the future. Create a separate variable for it.

File: .idea/gradle.xml [hunk 3]
```diff
@@ -11,6 +11,7 @@
           <set>
             <option value="$PROJECT_DIR$" />
             <option value="$PROJECT_DIR$/example" />
+            <option value="$PROJECT_DIR$/mylibrary" />
```
Comment (harsh-2711) [inline .idea/gradle.xml]: what is this `mylibrary` option?

**Review by harsh-2711 — CHANGES_REQUESTED** [review]

**Comment by harsh-2711** [comment]
@adit19shah Please do the suggested changes. We are then good to merge your PR.

File: searchwidget/src/main/res/values/colors.xml (hunk 1 above)
Comment (adit19shah) [reply searchwidget/src/main/res/values/colors.xml]: Got it, changed the color name to "black".

File: searchwidget/src/main/java/com/harsh/searchwidget/SearchBar.java (hunk 2 above)
Comment (adit19shah) [reply searchwidget/src/main/java/com/harsh/searchwidget/SearchBar.java]: Okay, got it, new variable added.

File: .idea/gradle.xml (hunk 3 above)
Comment (adit19shah) [reply .idea/gradle.xml]: Oh Sorry, this was accidently added by me while I was trying some other thing. I've removed it now.

**Comment by harsh-2711** [comment]
LGTM. Thanks for the PR.

File: settings.gradle [hunk 4]
```diff
@@ -1 +1,2 @@
+include ':mylibrary'
```
Comment (harsh-2711) [inline settings.gradle]:
```suggestion
```

File: example/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
+    <string name="search_placeholder">Search</string>
+
+    <string name="first_default_suggestion">Puma T-Shirt</string>
+    <string name="second_default_suggestion">Apple iPhone XS</string>
+    <string name="third_default_suggestion">Nike Trousers</string>
+
+    <string name="first_default_category">T-Shirt</string>
+    <string name="second_default_category">Mobiles</string>
+
+
```
Comment (harsh-2711) [inline example/src/main/res/values/strings.xml]: Extra line not needed.

**Review by harsh-2711 — CHANGES_REQUESTED** [review]
Apart from these changes, the rest LGTM

**Comment by harsh-2711** [comment]
LGTM. Thanks for the PR. From next time, please take a look at semantic commit messages guidelines that I posted on Gitter channel.

**Review by harsh-2711 — APPROVED** [review]

**Comment by harsh-2711** [comment]
give @adit19shah 5 points

**Pull request merged**